pkg/cmd/preview/options.go

```diff
@@ -72,14 +87,20 @@ func (o *PreviewOptions) Run() error {
 		DisableNone: o.DisableNone,
 		OverrideAST: o.OverrideAST,
 		NoStyle:     o.NoStyle,
+		NoPrompt:    o.Output == jsonOutput,
 	}, project, stack)
 	if err != nil {
 		return err
 	}
 
 	// return immediately if no resource found in stack
 	if sp == nil || len(sp.Resources) == 0 {
-		fmt.Println(pretty.GreenBold("\nNo resource found in this stack."))
+		if o.Output == jsonOutput {
+			emptyChanges, _ := json.Marshal(&opsmodels.Changes{})
```

Comment:
We can maintain the original output message to provide users with a friendly hint

Reply:
Maybe you can refer to the approach of Kubectl for non-existent resources. I think their approach is more user-friendly.

![image](https://github.com/KusionStack/kusion/assets/4793557/197b6da5-516d-41ed-a95c-471a8440b431)

Reply:
When errored, test-format output is shown; but if succeeded, the specified format is get output.

Reply:
The left feature upgrade is that even if there is an empty spec, diff should still be done, cause it means delete.